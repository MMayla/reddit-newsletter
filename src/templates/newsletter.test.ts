import pug from 'pug'
import path from 'path'
import fs from 'fs'

describe('Email Template', () => {
  it('Compile email template', () => {
    const compiledFunction = pug.compileFile(path.join(__dirname, 'newsletter.pug'))
    const html = compiledFunction({
      userFirstName: 'Mohamed',
      subs: {
        funny: [
          {
            title: 'title1',
            image: {
              url: 'https://preview.redd.it/rktwk0uop1n61.jpg?auto=webp&s=633eebc63d6cfafc30d4975abc82114fa41ea201',
            },
            votes: 300,
            url: 'https://www.reddit.com/r/funny/comments/m6fp40/i_guess_we_have_a_new_appliance/',
          },
          {
            title: 'title2',
            image: {
              url: 'https://preview.redd.it/rktwk0uop1n61.jpg?auto=webp&s=633eebc63d6cfafc30d4975abc82114fa41ea201',
            },
            votes: 45000,
            url: 'https://www.reddit.com/r/funny/comments/m6fp40/i_guess_we_have_a_new_appliance/',
          },
        ],
        notfunny: [
          {
            title: 'title3',
            votes: 780000,
            url: 'https://www.reddit.com/r/funny/comments/m6fp40/i_guess_we_have_a_new_appliance/',
          },
        ],
      },
    })

    // preview html
    fs.writeFileSync(path.join(__dirname, 'newsletter.html'), html)
    expect(html).toMatchSnapshot()
  })
})
