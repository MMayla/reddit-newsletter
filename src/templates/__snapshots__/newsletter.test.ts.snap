// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Template Compile email template 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><title>Reddit Newsletter</title><style>.wrapper {
    width: 80%;
    margin: auto;
  }
  
  .subreddit {
    margin-bottom: 80px;
  }
  
  .post {
    margin-bottom: 40px;
  }
  
  .center-text {
    text-align: center;
  }
  
  .tag {
    margin: auto;
    background-color: #FF5700;
    color: #fff;
    width: 4.5em;
    height: 2em;
    line-height: 2em;
    text-align: center;
    font-weight: bold;
    border-radius: 1em;
  }</style></head><body><h1>Hello Mohamed,</h1><p>See yesterdayâ€™s top voted posts from your favorite subreddits</p><div class=\\"wrapper\\"><div class=\\"subreddit\\"><div class=\\"center-text\\"><h2>funny</h2><p><a href=\\"https://www.reddit.com/r/funny\\" target=\\"_blank\\">https://www.reddit.com/r/funny</a></p></div><div class=\\"post\\"><img src=\\"https://preview.redd.it/rktwk0uop1n61.jpg?auto=webp&amp;s=633eebc63d6cfafc30d4975abc82114fa41ea201\\" alt=\\"title1\\" width=\\"100%\\"><div class=\\"tag\\">300</div><p class=\\"center-text\\"><a href=\\"https://www.reddit.com/r/funny/comments/m6fp40/i_guess_we_have_a_new_appliance/\\" target=\\"_blank\\">title1</a></p></div><div class=\\"post\\"><img src=\\"https://preview.redd.it/rktwk0uop1n61.jpg?auto=webp&amp;s=633eebc63d6cfafc30d4975abc82114fa41ea201\\" alt=\\"title2\\" width=\\"100%\\"><div class=\\"tag\\">45k</div><p class=\\"center-text\\"><a href=\\"https://www.reddit.com/r/funny/comments/m6fp40/i_guess_we_have_a_new_appliance/\\" target=\\"_blank\\">title2</a></p></div></div><div class=\\"subreddit\\"><div class=\\"center-text\\"><h2>notfunny</h2><p><a href=\\"https://www.reddit.com/r/notfunny\\" target=\\"_blank\\">https://www.reddit.com/r/notfunny</a></p></div><div class=\\"post\\"><div class=\\"tag\\">780k</div><p class=\\"center-text\\"><a href=\\"https://www.reddit.com/r/funny/comments/m6fp40/i_guess_we_have_a_new_appliance/\\" target=\\"_blank\\">title3</a></p></div></div></div></body></html>"
`;
